### ISSUE TITLE
Proselint doesn't work on readonly files

### ISSUE THREAD

**Issue by viccuad** [amperser/proselint#502]
On command_line.py:192
```python
            f = click.open_file(fp, 'r+', encoding="utf-8")
```

We are opening files to be linted with 'r+', which opens the file for both reading and writing (instead of 'r' which opens as read only).

This may not be needed, and breaks `proselint` for some files such as `demo.md` (therefore it breaks `proselint --demo`) when installed with a distribution package manager.

I've discovered this because of the Debian proselint package I mantain has that `demo.md`, and I have resorted on patching it temporarily to make it open files as readonly. Please, notify me if proselint intends to modify files in the future, and we need to find another solution.

**Comment by suchow** [comment]
Hm. What version of Proselint are you looking at? On master, the line you specified already uses `r`:

https://github.com/amperser/proselint/blob/master/proselint/command_line.py#L115

It seems this improvement was made by @Tatsh in commit a52c908890218b6ce2a2059bab5bbf634f883470.

**Comment by viccuad** [comment]
I found it on 0.5.3, the version that is currently [being shipped](https://tracker.debian.org/pkg/python3-proselint) in Debian.

I will happily drop the patch for the next shipped version :)

Seems like we can consider this fixed, let me close the issue.
Thanks and cheers.

**Comment by suchow** [comment]
@viccuad I've released Proselint 0.6.0, which includes this change.
